### PR TITLE
Docs urls in inspect output

### PR DESCRIPTION
**Pull request by simonbaird**

### CODE REVIEW

**Comment by simonbaird** [comment]
It's just the top commit. Draft because it should be rebased after the other PR is merged.

**Comment by simonbaird** [comment]
Rebased and moved out of draft.

File: internal/opa/output.go [hunk 1]
```diff
@@ -83,13 +85,22 @@ func prepTemplateFields(a *ast.AnnotationsRef) TemplateFields {
 		// conventions i.e. "package policy.<type>.<name>"
 		shortPath = trimmedPath
 	}
+
+	shortName := getShortName(a)
+
+	var docsUrl string
+	if len(pathStrings) > 4 && shortName != "" && (strings.HasPrefix(trimmedPath, "policy.pipeline.") || strings.HasPrefix(trimmedPath, "policy.release.")) {
+		docsUrl = fmt.Sprintf("https://hacbs-contract.github.io/ec-policies/%s_policy.html#%s__%s", pathStrings[2], pathStrings[3], shortName)
```
Comment (simonbaird) [inline internal/opa/output.go:93]: Later we could have a package annotation with the default docs url prefix. For now - hard coded.

File: internal/opa/output.go [hunk 2]
```diff
@@ -83,13 +85,22 @@ func prepTemplateFields(a *ast.AnnotationsRef) TemplateFields {
 		// conventions i.e. "package policy.<type>.<name>"
 		shortPath = trimmedPath
 	}
+
+	shortName := getShortName(a)
+
+	var docsUrl string
+	if len(pathStrings) > 4 && shortName != "" && (strings.HasPrefix(trimmedPath, "policy.pipeline.") || strings.HasPrefix(trimmedPath, "policy.release.")) {
```
Comment (lcarva) [inline internal/opa/output.go:92]: Why greater than 4?

**Review by lcarva — APPROVED** [review]

File: internal/opa/output.go (hunk 2 above)
Comment (simonbaird) [reply internal/opa/output.go:92]: Hmm yeah, probably should be `len(..) == 4`

Comment (simonbaird) [reply internal/opa/output.go:92]: Actually 5, because it still include "data" as element 0.

File: internal/opa/output.go [hunk 3]
```diff
@@ -47,12 +47,14 @@ type TemplateFields = struct {
...
 	ShortName   string
+	DocsUrl     string
 }
 
 var templates = map[string]string{
 	"text": hd.Doc(`
 		{{ .TrimmedPath }}.{{ .ShortName }} ({{ .WarnDeny }})
-		{{ .Title }}
+		{{ if .DocsUrl }}{{ .DocsUrl }}
+		{{ end }}{{ .Title }}
```
Comment (zregvart) [inline internal/opa/output.go:57]: Not important at all, just wanted to mention this is another way of doing this:
```suggestion
		{{ with .DocsUrl }}{{ . }}
		{{ end }}{{ .Title }}
```

File: internal/opa/output.go [hunk 4]
```diff
@@ -83,13 +85,22 @@ func prepTemplateFields(a *ast.AnnotationsRef) TemplateFields {
 		// conventions i.e. "package policy.<type>.<name>"
 		shortPath = trimmedPath
 	}
+
+	shortName := getShortName(a)
+
+	var docsUrl string
```
Comment (zregvart) [inline internal/opa/output.go:91]: It'd be nice to have the URL in Rego in a annotation, perhaps as a followup...

**Review by zregvart — APPROVED** [review]

File: internal/opa/output.go (hunk 2 above)
Comment (lcarva) [reply internal/opa/output.go:92]: Ah so this is checking if it includes a certain prefix?. If so, maybe something like this is more easily read: `len(pathStrings) > len("myprefix")`. Or even better `strings.HasPrefix(pathStrings, "myprefixX")`, if `X` is always the same value.

File: internal/opa/output.go (hunk 3 above)
Comment (simonbaird) [reply internal/opa/output.go:57]: I like it.

File: internal/opa/output.go (hunk 4 above)
Comment (simonbaird) [reply internal/opa/output.go:100]: Agreed. I added some commentary along those lines.

**Comment by simonbaird** [comment]
Thanks for the reviews. I made a few tweaks. Will stop fussing with this and probably merge when it goes green.

**Pull request merged**